pkg/storaged/fsys-tab.jsx

```diff
@@ -18,24 +18,279 @@
  */
 
 import React from "react";
+import { Alert } from "@patternfly/react-core";
 
 import cockpit from "cockpit";
 import * as utils from "./utils.js";
 
-import { dialog_open, TextInput } from "./dialog.jsx";
+import { dialog_open, TextInput, CheckBoxes } from "./dialog.jsx";
 import { StorageButton, StorageLink } from "./storage-controls.jsx";
-import * as FormatDialog from "./format-dialog.jsx";
+import { initial_tab_options, parse_options, unparse_options, extract_option } from "./format-dialog.jsx";
 
 const _ = cockpit.gettext;
 
+export function is_mounted(client, block) {
+    const block_fsys = client.blocks_fsys[block.path];
+    const mounted_at = block_fsys ? block_fsys.MountPoints : [];
+    const config = utils.array_find(block.Configuration, function (c) { return c[0] == "fstab" });
+    if (config && config[1].dir.v) {
+        let dir = utils.decode_filename(config[1].dir.v);
+        if (dir[0] != "/")
+            dir = "/" + dir;
+        return mounted_at.map(utils.decode_filename).indexOf(dir) >= 0;
+    } else
+        return null;
+}
+
+export function get_fstab_config(block) {
+    const config = utils.array_find(block.Configuration, function (c) { return c[0] == "fstab" });
+    if (config) {
+        let dir = utils.decode_filename(config[1].dir.v);
+        const opts = (utils.decode_filename(config[1].opts.v)
+                .split(",")
+                .filter(function (s) { return s.indexOf("x-parent") !== 0 })
```

Comment:
This isn't new code, but I wonder anyway: systemd is entirely ignorant of the x-parent option. udisks seems to read and/or write it, but that's not involved at boot. What does that option do in practice?

Reply:
UDisks2 uses this to clean out fstab when deleting nested things like logical volumes and LUKS devices.  With the x-parent annotation, this even works when we can't read the contents of those things, like when removing a locked LUKS device without knowing the passphrase.